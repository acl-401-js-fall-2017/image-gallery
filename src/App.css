.App {
  text-align: center;
}

.completed {
  text-decoration: line-through;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40px;
}

.App-header {
  background-color: steelblue;
  height: 120px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

nav {
  margin: 10px 0;
}
nav > li {
  display: inline-block;
  margin: 0 10px;
}

.nav-link {
  text-decoration: none;
  padding: 5px 10px;
}

.nav-link:visited {
  color: white;
}

.nav-link.active {
  border: 1px solid white;
}

.items {
  margin: 0;
  padding: 0;
  margin-bottom: 25px;
}

.items > li {
  list-style: none;
  margin: 5px 0;
  padding: 10px;
  width: 250px;
  margin-right: auto;
  margin-left: auto;
  text-align: left;
  background: rgba(249, 171, 77, 0.5);
}

.items > li:hover {
  background: rgba(249, 171, 77, .8);
  color: white;
}

.items > li > button {
  float: right;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.list-view {
  display: flex;
  justify-content: space-evenly;
  padding: 20px;
}