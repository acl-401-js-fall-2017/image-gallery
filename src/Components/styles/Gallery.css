.Gallery {
    height: 100%;
}

.imgSlide {
    height: 60%;
    width: 80%;
    margin: 1em auto;
    overflow: hidden;
}

.Gallery img {
    height: 90%;

}

.galleryButtons {
    display: flex;
    justify-content: center;
    height: 10%;
}

.galleryButtons h1 {
    width:50%;
    text-align: center;
    background-color: black;
    border-radius: 0.25em;
    color: white;
    margin: 0.75em 1em;
    line-height: 1.5em;
}