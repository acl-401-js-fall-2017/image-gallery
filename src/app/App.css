body {
  margin:10%;
}
.topnav {
  background-color: #333;
  overflow: hidden;
  display: block;
}

.nav  {
  float: left;
  display: block;
  color: #979494;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
  font-size: 17px;
}
.nav:hover {
  background-color: #ddd;
  color: black;
}
.nav.active {
  background-color: #4CAF50;
  color: white;
}

.home {
  display: inline-block;
  color: #7A7979;
  text-align: left;
  margin:8%;
  font-size: 22px;
}

.albums {
  display: inline;
  color: #7A7979;
  text-align: left;
  padding-left: 20%;
  margin:8%;
  font-size: 22px;
}
.albums h4 {
  margin-top:10%;
  margin-left:8%;
  font-size: 24px;
}

.albums ul {
  border: .5px dotted #ADA8A8;
  background-color: #EEEEEE;
  margin-left:8%;
  margin-right:8%;
  border-radius: 8px;
}
.albums li {
  display: block;
  width: 80%;
  color: #979494;
  text-align: left;
  line-height: 1px;
  padding: 0.8%;
  font-size: 20px;
}

 
/* .needSpace{
  width:800px;
} */

.add-album {
  margin-left:8%;
}

#thumbnail-view{
  margin-right:200px;
  padding-top:200px;
  display: flex;
  justify-content: space-around;
}

.flexy{
  justify-content: space-around;
}

