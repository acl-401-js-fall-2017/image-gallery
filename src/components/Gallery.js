import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import Bunnies from '../images/bunnies';



export default class Gallery extends PureComponent {

    static propTypes = {
      bunnies: PropTypes.arrayOf(PropTypes.object).isRequired
    }

    constructor(){
      super();
      this.state = {
        bunnies: Bunnies,
        picIndex: 0
      };
    }


    nextImage = ({ target }) => {
      this.setState(prevState => {
        let picIndex;
        if (target.id === 'prevButton') {
          picIndex = prevState.picIndex - 1;
        }
        if (target.id === 'nextButton') {
          picIndex = prevState.picIndex + 1;
        }
        return { picIndex };
      });
    }

    render() {
      const bunny = this.props.bunnies[this.state.picIndex];

      return (
        <div className="gallery">
          <img src={bunny.url} alt={bunny.title} height="100px"/>
          <h4>{bunny.title}</h4>
          <h6>{bunny.description}</h6>

          {this.state.picIndex !== 0 && (
            <button type="button" value="Previous" id="prevButton" onClick={this.nextImage}>
          Previous
            </button>
          )}

          {this.state.picIndex !== this.state.bunnies.length - 1  && (
            <button type="button" value="Next" id="nextButton" onClick={this.nextImage}>
          Next
            </button>
          )}
          {/* Might want to add remove button here */}
        </div>
      );
    }
}