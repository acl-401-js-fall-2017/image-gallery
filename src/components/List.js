import React, { Component } from 'react';
import PropTypes from 'prop-types';

import AddImage from './AddImage';

export default class List extends Component {
   
    static propTypes = {
      images: PropTypes.arrayOf(PropTypes.object).isRequired
    }


    render() {
      const { addImage, handleRemove } = this.props;
      return (
        <div className="list-view">
          <table>
            <tbody>
              {this.props.images.map((image, index) => {
                return (
                  <tr key={index}>
                    <td><a href={image.url}>{image.title}</a></td>
                    <td>{image.description}</td>
                    <td> <input type="button" data-value={image._id} value="remove" onClick={() => handleRemove(image._id)}/></td>
                  </tr>
                );
              })}
              <tr>
                <td></td>
                <td><AddImage addImage={newImage => addImage(newImage)}/></td>
                <td></td>
              </tr>
            </tbody>
          </table>
        </div>
      );
    }
}