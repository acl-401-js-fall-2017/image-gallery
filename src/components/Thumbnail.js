import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class Thumbnail extends PureComponent {
    
    static propTypes = {
      bunnies: PropTypes.arrayOf(PropTypes.object).isRequired
    }


    render() {
      return (
        <div className="thumbnail">
          {this.props.bunnies.map((bunny, index) => {
            return (
              <div key={index}>
                <img src={bunny.url} alt={bunny.title} height="100px"/>
                <h4>{bunny.title}</h4>
                <h6>{bunny.description}</h6>
              </div>
            );
          })}
        </div>
      );
    }
}