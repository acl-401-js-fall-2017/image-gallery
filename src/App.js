import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Redirect, Switch } from 'react-router-dom';

import './App.css';

import HeaderRoutes from './components/HeaderRoutes';
import Home from './components/Home';
import About from './components/About';
import Albums from './components/Albums';
import Album from './components/Album';


class App extends Component {

  render() {
    return (
      <Router>
        <div className="App">
          <HeaderRoutes/>
          <Switch>
            <Route exact path="/" component={Home}/>
            <Route exact path="/albums" component={Albums}/>
            <Route exact path="/albums/:id/(gallery|thumbnail|list)?" component={Album}/>
            <Route exact path="/about" component={About}/>
            <Redirect to="/"/>
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;
