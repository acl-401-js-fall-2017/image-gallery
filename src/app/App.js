import React, { Component } from 'react';
import { addAlbum } from '../action-albums';
import { removeAlbum } from '../action-albums';
import './App.css';
import View from '../views/View';
import Albums from '../views/Albums';
import { 
  BrowserRouter as Router, 
  Route, Switch, Redirect } from 'react-router-dom';
import Header from './Header';
	
class App extends Component {	
  constructor() {
    super();
    this.state = {
      albums: []
    };
  }

  async componentDidMount() {
    const albums = await fetch('/api/albums').then(response => response.json());
    const newState = { ...this.state, albums };
    this.setState(newState);
  }

  handleAddAlbum = async(album) => {
    const albumName = await fetch('/api/albums', {
      method: 'post',
      body: JSON.stringify(album),
      headers: {
        'Accept': 'application/json',
        'Content-type': 'application/json'
      }
    }).then(response => response.json());
    const newState = addAlbum(this.state, albumName);
    this.setState(newState);
  }

  handleRemoveAlbum = async(id) => {
    await fetch(`/api/albums/${id}`, {
      method: 'delete',
    }).then(response => response.json());
    const newState = removeAlbum(this.state, id);
    this.setState(newState);
  }

  render(){
    const MyAlbums = () => {
      return (
        <Albums 
          onRemove={this.handleRemoveAlbum.bind(this)}
          onAddAlbum={this.handleAddAlbum.bind(this)}
          albums={this.state.albums}
        />
      );
    };
    return(
      <Router>
        <div>
          <Header/>
          <Switch>
            <Route exact path="/" component={Home}/>
            <Route exact path="/about" component={About}/>
            <Route exact path="/albums" render={MyAlbums}/>
            <Route path="/albums/:albumId" component={View}/>
            <Redirect to="/"/>
          </Switch>
        </div>
      </Router>
    );
  }
}
export default App;

const Home = () => (
  <div id="intro">
    <p>Welcome to the bunny page.</p>
  </div>
);

const About = () => (
  <div id="about">
    <h1>About Bunnies</h1>
    <p>Bunnies are spectacularly cute. But did you know that when a home houses 50 bunnies, the home is spectacularly smelly?</p>
  </div>
);