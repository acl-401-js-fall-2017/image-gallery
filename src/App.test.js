import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import Gallery from './Gallery';
import Thumbnail from './Thumbnail';
import List from './List';
import View from './View';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('renders a snapshot of View', () => {
  const tree = renderer.create(<View/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it ('renders a snapshot of List', () => {
  const tree = renderer.create(<List bunnies={[]} onRemove={() =>{}} onAddImg={() =>{}}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it ('renders a snapshot of Thumbnail', () => {
  const tree = renderer.create(<Thumbnail bunnies={[]}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it ('renders a snapshot of Gallery', () => {
  const bunny = {
    _id: 3454,
    title: 'Harlequin Bunny',
    description: 'This harlequin bunny artfully poses for a picture.',
    url: 'http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg',
  };
  const tree = renderer.create(<Gallery bunnies={[bunny]}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
<<<<<<< HEAD
=======

>>>>>>> upstream/mugworks

