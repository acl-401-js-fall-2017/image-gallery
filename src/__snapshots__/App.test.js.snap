// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of Gallery 1`] = `
<div
  id="gallery-container"
>
  <button
    className="button previous"
    disabled={true}
    onClick={[Function]}
  >
    ←Previous
  </button>
  <section>
    <img
      alt="bunny"
      src="http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg"
    />
    <figcaption>
      Harlequin Bunny
    </figcaption>
    <figcaption>
      This harlequin bunny artfully poses for a picture.
    </figcaption>
  </section>
  <button
    className="button next"
    disabled={false}
    onClick={[Function]}
  >
    Next→
  </button>
</div>
`;

exports[`renders a snapshot of List 1`] = `
<div>
  <table
    id="table"
  >
    <thead>
      <tr>
        <th>
          Title
        </th>
        <th>
          Description
        </th>
        <th />
      </tr>
    </thead>
    <tbody />
  </table>
  <form
    id="form"
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        Add an image
      </legend>
      <label>
        Image Title: 
      </label>
      <input
        id="form-input"
        name="title"
      />
      <label>
        Description: 
      </label>
      <input
        id="form-input"
        name="description"
      />
      <br />
      <label>
        Url: 
      </label>
      <input
        id="form-input"
        name="url"
      />
      <button
        className="button add"
        type="submit"
      >
        Add
      </button>
    </fieldset>
  </form>
</div>
`;

exports[`renders a snapshot of Thumbnail 1`] = `
<div
  className="thumbnail"
/>
`;
